test/org/sagebionetworks/bridge/play/controllers/ConsentControllerMockedTest.java

```diff
@@ -163,8 +168,11 @@ public void canWithdrawConsent() throws Exception {
         JsonNode node = BridgeObjectMapper.get().readTree(response);
         assertEquals("User has been withdrawn from the study.", node.get("message").asText());
         
+        // Should call the service and withdraw
         verify(consentService).withdrawConsent(any(Study.class), any(User.class), captor.capture(), any(Long.class));
         assertEquals("Because, reasons.", captor.getValue().getReason());
+        
+        verify(cacheProvider).setUserSession(session);
```

Comment:
Not sure this does what you think it does. You have two cacheProviders, one at the class level, one local to the method. This is a little confusing.